packages/vats/src/proposals/upgrade-board-proposal.js

```diff
@@ -0,0 +1,50 @@
+import { E } from '@endo/far';
+
+/**
+ * @param {BootstrapPowers & {
+ *   consume: {
+ *     vatAdminSvc: VatAdminSvc;
+ *     vatStore: MapStore<
+ *       string,
+ *       import('@agoric/swingset-vat').CreateVatResults
+ *     >;
+ *   };
+ * }} powers
+ * @param {object} options
+ * @param {{ boardRef: VatSourceRef }} options.options
+ */
+export const upgradeBoard = async (
+  { consume: { board, vatAdminSvc, vatStore } },
+  options,
+) => {
+  const { boardRef } = options.options;
+
+  assert(boardRef.bundleID);
+  const boardBundleCap = await E(vatAdminSvc).getBundleCap(boardRef.bundleID);
+  console.log(`Board BUNDLE ID: `, boardRef.bundleID);
+  const idsBefore = await E(board).ids();
+
+  const { adminNode } = await E(vatStore).get('board');
+  await E(adminNode).upgrade(boardBundleCap, {});
+
+  const idsAfter = await E(board).ids();
+  const same =
+    idsBefore.length === idsAfter.length &&
+    idsBefore.every((element, index) => element === idsAfter[index]);
+  assert(same, 'keys must stay the same');
```

Comment:
Is this a test assertion? Or is it intended to run in production?

If it fails, we know there's a problem, but the damage is already done, right?

Reply:
The same proposal with this assertion will run in production as on our test networks. If it succeeds in tests, we'll have confidence that it will succeed in production.

Would you suggest logging the result rather than asserting it?

Reply:
So it's in prod by design. Ok.